paper/JMNotes.Rmd

```diff
@@ -0,0 +1,45 @@
+# remarks JM
+
+- abstract: it is at least uncommon to cite other works in the abstract.
+If not explicitly required by the journal, I would refrain from it since you most likely cite all these works in the paper itself.
+- The last sentence of the abstract does not add much.
+Delete it or rewrite in a more compelling way.
+
+- I like the crop classification example, however, it is not entirely clear to me what agricultural districts are. 
+And though the crop classification introduces non-standard spatial resampling strategies, it does not make reference to the more common spatial resampling procedure (k-means clustering) where clustered coordinates are considered, which might be a more gentle introduction to spatial CV for readers who are not already familiar with the concepts.
+But I might be wrong.
+Looking at Table 1, the spatial blocking/tiles approach seems to be even more popular than the kmeans clustering approach.
+
+- Table 1: Wow, not too many papers have so far used spatial CV.
+Is this a comprehensive list?
+Have you somewhere listed the references by chosen strategy?
+If so, put it in the Appendix and reference the Appendix in the caption.
+Not entirely clear, what the footnotes refer to.
+These seem to be studies which have used the corresponding spatial CV approach but it is not all of them (e.g., spcv_block has been used by 28 papers but only 3 are referenced). 
+Please clarify.
+
+- I really do like the visualizations, great job! And they help the reader a lot to better grasp the differences between the resampling strategies.
+
+- In case, you didn't know: from 1.0.0 onwards, sf uses s2 by default, i.e. spherical geometries.
+Hence, I am not sure if there is really much more work to do (see https://r-spatial.github.io/sf/articles/sf7.html).
+I am referring to this sentence in the paper:
+
+>It is planned to provide a more consistent interface that supports all CRS and allows the use of distance measures in projected planar space or using great circle distances on the unprojected ellipsoid, in the case of unprojected CRS.
+
+- The leave-one-disc-out method seems to be quite promising since the direct neighbors between the training and test sets are removed thus removing spatial autocorrelation which might still be present when using the kmeans clustering approach.
+In the back of my mind, I knew that this was possible but I didn't know that this was already implemented in **sperrorest**.
+I wonder why AB has often preferred the kmeans clustering approach over the leave-one-disc-out approach.
+Maybe something which can be discussed.
```

Comment:
LODO-CV:

* Yes, sperrorest does support LODO with and without a buffer.
* Remember you must fit a model on each training sample. When discs are small (extreme case: size 1 as in standard LOO), this means that a model is fitted for making only one prediction. This is computationally extremely expensive.
* When buffer size is large relative to the size of the study area, a large percentage of the study is not being sampled.
* Patrick, avoid claims of one method being better or less biased unless a clear concept of bias is associated with these claims. The questions are: 1. What do I want to estimate (i.e. what is appropriate for the application at hand. 2. How can I estimate it. - There is no bias in 2. as long as there's no answer to 1. I think I tried to clarify this in my background section, but I'll double-check this in the next pass. Overall, I think it is beyond the scope of the paper to make specific recommendations to use one or another method. We just don't know what people want to do in their analysis. The only thing mlr3 offers is an estimation procedure.
* This being said, I have something coming up that involves spLOO (and by extension, LODO) and comparison with kmeans...

Reply:
Excellent, thanks for the clarification!

Reply:
Thanks Alex.

> Patrick, avoid claims of one method being better or less biased unless a clear concept of bias is associated with these claims.
> Overall, I think it is beyond the scope of the paper to make specific recommendations to use one or another method. 

Yes, I am trying to avoid that. Please highlight parts in which it might read like this.
It is not always easy to avoid that feeling with certain wordings but this is really an important point to get right here.

> This being said, I have something coming up that involves spLOO (and by extension, LODO) and comparison with kmeans...

Looking forward!

---


@alexanderbrenning 

A very specific point that would be of interest for me here (and maybe this is also partly what Jannes was referring to):

Why not choose `spcv_disc` with a "not too large" buffer (TBD how to determine the buffer size) over `spcv_coords` in scenarios where both would be appropriate? I.e. in an irregularly shaped study area where we would like to use a 5-10 fold CV? With a somewhat small buffer one should be able to reduce SAC at least a bit between train and test.

In most of your past studies you went with `spcv_coords` so it would be interesting to hear your thoughts about this :)